tests/php/Security/GroupTest.php

```diff
@@ -171,6 +171,60 @@ public function testGetAllChildren()
         $this->assertSame(['childgroup', 'grandchildgroup'], $children->column('Code'));
     }
 
+    public function testGroupInGroupMethods()
+    {
+        $parentGroup = $this->objFromFixture(Group::class, 'parentgroup');
+        $childGroup = $this->objFromFixture(Group::class, 'childgroup');
+        $grandchildGroup = $this->objFromFixture(Group::class, 'grandchildgroup');
+        $adminGroup = $this->objFromFixture(Group::class, 'admingroup');
+        $group1 = $this->objFromFixture(Group::class, 'group1');
+
+        $this->assertTrue($grandchildGroup->inGroup($childGroup));
+        $this->assertTrue($grandchildGroup->inGroup($childGroup->ID));
+        $this->assertTrue($grandchildGroup->inGroup($childGroup->Code));
+
+        $this->assertTrue($grandchildGroup->inGroup($parentGroup));
+        $this->assertTrue($grandchildGroup->inGroups([$parentGroup, $childGroup]));
+        $this->assertTrue($grandchildGroup->inGroups([$childGroup, $parentGroup]));
+        $this->assertTrue($grandchildGroup->inGroups([$parentGroup, $childGroup], true));
+
+        $this->assertFalse($grandchildGroup->inGroup($adminGroup));
+        $this->assertFalse($grandchildGroup->inGroups([$adminGroup, $group1]));
+        $this->assertFalse($grandchildGroup->inGroups([$adminGroup, $childGroup], true));
+
+        // Edgecases
+        $this->assertTrue($grandchildGroup->inGroup($grandchildGroup));
+        $this->assertFalse($grandchildGroup->inGroups([]));
+        $this->assertFalse($grandchildGroup->inGroups([], true));
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     */
+    public function testGroupInGroupExceptionForFakeGroup()
+    {
+        $group = new Group();
+        $group->inGroup('NotARealGroup');
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     */
+    public function testGroupInGroupExceptionForFakeGroupID()
+    {
+        $group = new Group();
+        $group->inGroup(99999999);
```

Comment:
I think it should be an exception, but the wrong exception message is thrown. It says "invalid type" when the type is valid, but the ID isn't a real record.